assignment5/assignment5.py

```diff
@@ -0,0 +1,45 @@
+class AlienDictionary(object):
+    def alienOrder(self, words):
+        # Find ancestors of each node by DFS.
+        nodes, ancestors = sets.Set(), {}
+        for i in xrange(len(words)):
+            for c in words[i]:
+                nodes.add(c)
+        for node in nodes:
+            ancestors[node] = []
+        for i in xrange(1, len(words)):
+            if len(words[i-1]) > len(words[i]) and \
```

Comment:
I think we assume "empty" character sorts before real character. In that case `words[i-1]` should be the shorter one to conceivably be a full prefix of `words[i]`.